File: cc/FactoryMethod/Base/inputclient.cc
#include "input.h"
#include "inputfactory.h"

int main(int argc, char* argv[])
{
    InputFactory* factory = new FileInputFactory();
    Input* input = factory->create();
    input->read();
    delete input;
    delete factory;

    factory = new ConsoleInputFactory();
    input = factory->create();
    input->read();
    delete input;
    delete factory;

    factory = new NetInputFactory();
    input = factory->create();
    input->read();
    delete input;
    delete factory;

    return 0;
}

File: python/FactoryMethod/inputclient.py
"""
inputclient.py
"""

from input import *
from inputfactory import *

def main():
    factory = FileInputFactory()
    input = factory.create()
    input.read()

    factory = ConsoleInputFactory()
    input = factory.create()
    input.read()

    factory = NetInputFactory()
    input = factory.create()
    input.read()

if __name__ == '__main__':
    main()

